File: code/streaming-jobs/word-count/reducer.py
#!/usr/bin/env python2

import fileinput

current_key = None
current_sum = 0

for line in fileinput.input():
	key, count = line.split()
	
	if key == current_key:
		current_sum += int(count)
	else:
		if current_key != None:
			print '{}\t{}'.format(current_key, current_sum)	
		current_key = key
		current_sum = int(count)

if current_key != None:
	print '{}\t{}'.format(current_key, current_sum)

File: code/streaming-jobs/word-count/reducer.php
#!/usr/bin/php
<?php

$last_word = "";
$last_count = 0;

while (($line = fgets(STDIN)) !== false) {
    $words = explode("\t", trim($line)); 
    $word = $words[0];
    $count = intval($words[1]);

    if($word == $last_word) {
        $last_count += $count;
    } else {
        if ($last_word != "") {
            echo "$last_word\t$last_count\n";
	}
        $last_word = $word;
	$last_count = $count;
    }
}//end while
        if ($last_word != "") {
            echo "$last_word\t$last_count\n";
	}
?>
